service/nodeconfig/v1/resource/node/create.go

```diff
@@ -37,7 +37,15 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "looking for certificates for the guest cluster")
 
 		draining, err = r.certsSearcher.SearchDraining(key.ClusterID(customObject))
-		if err != nil {
+		if certs.IsTimeout(err) {
+			// Here we log a warning for alerting purposes and also return an error to
+			// make the resource execution being retried. Then the amount of warning
+			// logs will surge and we have a chance to try to drain again in case
+			// there are only some weird connection issues to the guest cluster
+			// Kubernetes API.
+			r.logger.LogCtx(ctx, "level", "warning", "message", "cannot find certificates for guest cluster '%s'", key.ClusterID(customObject))
```

Comment:
I think the warning could be improved, maybe mention that this happened because of a timeout?

Reply:
Not convinced. How would this improve the message? Is this an important information though? I was thinking about adding something to the real context that draining doesn't work because of that but I couldn't come up with something sexy.

Reply:
True, it doesn't really help since the error is returned aswell. I can't come up with something better either.

Reply:
Just added `draining` to the log message.